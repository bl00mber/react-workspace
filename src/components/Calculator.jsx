import React from 'react';
import PropTypes from 'prop-types';

import TemperatureInput from './TemperatureInput';
import BoilingVerdict from './BoilingVerdict';

const Calculator = props => (
  <div>
    <TemperatureInput
      scale="c"
      temperature={props.celsius}
      onTemperatureChange={temperature => props.onCelsiusChange(temperature)}
    />

    <TemperatureInput
      scale="f"
      temperature={props.fahrenheit}
      onTemperatureChange={temperature => props.onFahrenheitChange(temperature)}
    />

    <BoilingVerdict
      celsius={parseFloat(props.celsius || 0)}
    />

  </div>
);

Calculator.propTypes = {
  celsius: PropTypes.string.isRequired,
  fahrenheit: PropTypes.string.isRequired,
  onCelsiusChange: PropTypes.func.isRequired,
  onFahrenheitChange: PropTypes.func.isRequired,
};

export default Calculator;
