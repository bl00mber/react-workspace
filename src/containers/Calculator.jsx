import { connect } from 'react-redux';

import CalculatorComponent from '@/components/Calculator';
import { setCelsius, setFahrenheit } from '@/store/actions';

const mapStateToProps = state => ({
  celsius: state.temperatures.celsius,
  fahrenheit: state.temperatures.fahrenheit,
});

const mapDispatchToProps = dispatch => ({
  onCelsiusChange: temperature => dispatch(setCelsius(temperature)),
  onFahrenheitChange: temperature => dispatch(setFahrenheit(temperature)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(CalculatorComponent);
