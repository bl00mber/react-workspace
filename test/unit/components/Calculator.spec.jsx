import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Calculator from '@/components/Calculator';

describe('Calculator', () => {
  test('should match enzyme shallow snapshot', () => {
    const wrapper = shallow(<Calculator />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
