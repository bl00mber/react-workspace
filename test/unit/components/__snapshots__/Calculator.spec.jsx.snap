// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should match enzyme shallow snapshot 1`] = `
<div>
  <TemperatureInput
    onTemperatureChange={[Function]}
    scale="c"
    temperature="celsius"
  />
  <TemperatureInput
    onTemperatureChange={[Function]}
    scale="f"
    temperature="fahrenheit"
  />
  <BoilingVerdict
    celsius={0}
  />
</div>
`;
